api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -120,32 +119,32 @@ type NodeGroup struct {
 	Config *NodeGroupConfig `json:"config,omitempty"`
 }
 
+// NodeGroupStatus defines the status of a single node group
+type NodeGroupStatus struct {
+	// DaemonSet of the configured RTE for this node group
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="RTE DaemonSet"
+	DaemonSet NamespacedName `json:"daemonsets,omitempty"`
+	// NodeGroupConfig represents the latest available configuration applied to this NodeGroup
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Optional configuration enforced on this NodeGroup"
+	Config *NodeGroupConfig `json:"config,omitempty"`
+	// MachineConfigPoolName matches the name of the MCP derived from the MCP selector
+	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="Matching MCP name for this node group"
+	MachineConfigPoolName string `json:"machineConfigPoolName"`
+}
+
 // NUMAResourcesOperatorStatus defines the observed state of NUMAResourcesOperator
 type NUMAResourcesOperatorStatus struct {
-	// DaemonSets of the configured RTEs, one per node group
-	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="RTE DaemonSets"
-	DaemonSets []NamespacedName `json:"daemonsets,omitempty"`
-	// MachineConfigPools resolved from configured node groups
-	//+operator-sdk:csv:customresourcedefinitions:type=status,displayName="RTE MCPs from node groups"
-	MachineConfigPools []MachineConfigPool `json:"machineconfigpools,omitempty"`
-	// Conditions show the current state of the NUMAResourcesOperator Operator
```

Comment:
we can't remove fields from the API until we bump the version. The best we can do is to not set any values to them. In this case we need to keep setting these values for backward compatibility.

Reply:
I appreciate your comment. These are in the operator status which the operator user does not use to configure the operator. I have updated the conversion between the API versions to match what's needed in the new fields. Do you think these modifications are not enough to use the new instance of the version? 
It will help if you could provide an example on how the compatibility is affected by this change